examples/hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/ui/GenerateDialog.java

```diff
@@ -231,4 +185,64 @@ public boolean equals(final Object obj) {// NOSONAR - as this is generated
         void okButton(final String namePrefix, final String tenant, final int amount, final int pollDelay,
                 final URL basePollURL, final String gatewayToken, final Protocol protocol);
     }
+
+    private OptionGroup createProtocolGroup() {
+
+        final OptionGroup protocolGroup = new OptionGroup("Simulated Device Protocol");
+        protocolGroup.addItem(Protocol.DMF_AMQP);
+        protocolGroup.addItem(Protocol.DDI_HTTP);
+        protocolGroup.setItemCaption(Protocol.DMF_AMQP, "Device Management Federation API (AMQP push)");
+        protocolGroup.setItemCaption(Protocol.DDI_HTTP, "Direct Device Interface (HTTP poll)");
+        protocolGroup.setNullSelectionAllowed(false);
+        protocolGroup.select(Protocol.DMF_AMQP);
+        protocolGroup.addValueChangeListener(event -> {
+            pollDelayTextField.setVisible(!pollDelayTextField.isVisible());
```

Comment:
The behavior is still the same, but nevertheless thanks for the commit. With that approach it is clearer why visibility is changing.